packages/pharos/src/components/popover/pharos-popover.ts

```diff
@@ -0,0 +1,363 @@
+              offset(8),
+              flip({
+                fallbackPlacements: this._filteredFallbackPlacements,
+              }),
+            ],
+          }).then(({ x, y }) => {
+            Object.assign(this._popover.style, {
+              left: `${x}px`,
+              top: `${y}px`,
+            });
+          });
+        }
+      });
+    }
+  }
+
+  private async _handleTriggerClick(event: MouseEvent): Promise<void> {
+    const trigger = event.currentTarget as Element;
+    const otherTriggerClicked = this._currentTrigger && this._currentTrigger !== trigger;
+    this._currentTrigger = trigger;
+
+    if (this._hasHover && this.open) {
+      return;
+    } else if (!otherTriggerClicked) {
+      this.open = !this.open;
+    } else {
+      this.open = false;
+      await this.updateComplete;
+      debounce(() => {
+        this._currentTrigger = trigger;
+        this.open = true;
+      }, 150)();
+    }
+  }
+
+  private _handleTriggerKeydown(event: KeyboardEvent): void {
+    switch (event.key) {
+      case 'Enter':
+      case ' ':
+      case 'Spacebar':
+        if ((event.target as Element)?.hasAttribute('data-popover-hover')) {
+          event.preventDefault();
+          this._openPopover(event);
+        }
+        break;
+    }
+  }
+
+  private async _handleTriggerHover(event: MouseEvent): Promise<void> {
+    if (event.type === 'mouseenter') {
+      const otherTriggerClicked = this._currentTrigger && this._currentTrigger !== event.target;
+      this._currentTrigger = event.target as Element;
+      this._hasHover = true;
+
+      if (otherTriggerClicked) {
+        this.open = false;
+        await this.updateComplete;
+        this._currentTrigger = event.target as Element;
+      }
+    } else if (event.type === 'mouseleave') {
+      this._hasHover = false;
+    }
+    debounce(() => {
+      this._setOpen();
+    }, 150)();
+  }
+
+  private _handleKeydown(event: KeyboardEvent): void {
+    if ((event.key === 'Escape' || event.key === 'Esc') && this.open) {
+      event.stopPropagation();
+      this.open = false;
+    }
+  }
+
+  private _handleClick(event: MouseEvent): void {
+    const targetClicked = this._triggers.find(
+      (trigger) => trigger === (event.target as Element)?.closest(trigger.tagName)
+    );
+    const popoverClicked =
+      this === (event.target as Element)?.closest('[data-pharos-component="PharosPopover"]');
+    if (!targetClicked && !popoverClicked && this.open) {
+      event.stopPropagation();
+      this.open = false;
+    }
+  }
+
+  private _setOpen(): void {
+    this.open = this._hasHover;
+  }
+
+  private _handleHover(event: MouseEvent): void {
+    if (event.type === 'mouseenter') {
+      this._hasHover = true;
+    } else if (event.type === 'mouseleave') {
+      this._hasHover = false;
+    }
+    debounce(() => {
+      this._setOpen();
+    }, 150)();
+  }
+
+  private async _focusContents(): Promise<void> {
+    // TODO: Probably focus something
```

Comment:
still something to do here?

Reply:
Good catch! Added as part of https://github.com/ithaka/pharos/pull/588/commits/9c41eb81b8d1c2d9fa44451310c3ebfed0f117ff